### PR TITLE
Introducing Docker to Grape

### PR DESCRIPTION
**Pull request by ericproulx**
Hi,

I've been using Docker for a while now and I thought it would be a great addition to include a Dockerfile to Grape. Also, `docker-compose` always been my go to for running containers even though its for multiple containers. It's just easier to configure than running `docker` commands with arguments.

## Dockerfile
- As an `ARG` called `RUBY_VERSION` for building an image on different ruby-version
- Setting `BUNDLE_PATH` to `/usr/local/bundle/gems`
- Setting `LIB_PATH` to `/var/grape`. Code from the host will be copied in that folder
- Add basic packages for development
- Install latest version of `rubygems` and `bundle`
- Starts with an entrypoint

## Entrypoint
- echoes version of ruby, rubygems and bundle.
- check gems sanity with bundle
- update gems to latest version
- execute `bundle exec :your_command`

## Docker-compose
- Default `RUBY_VERSION` to `3`. 
- Add `gems` volume

## Basic CLI
- Basic command  `docker-compose run --rm --build grape <your_command>`. You may replace `<your_command>` by anything that runs under `bundle exec`
- `--rm` will remove automatically the container
- `--build` will automatically build or rebuild the image (useful when changing RUBY_VERSION)

## CLI
- Running all specs `docker-compose run --rm --build grape rspec`
- Running rspec on a specific file `docker-compose run --rm --build grape rspec spec/:file_path`
- Running task `docker-compose run --rm --build grape rake <task_name>`
- Running rubocop `docker-compose run --rm --build grape rubocop`
- Running all specs on a specific ruby version (e.g 2.7.7) `RUBY_VERSION=2.7.7 docker-compose run --rm --build grape rspec`
- Running specs on a specific gemfile (e.g rails_7.gemfile) `docker-compose run -e GEMFILE=rails_7 --rm --build grape rspec`

Running the container will automatically install the missings gems and run your command.

### CODE REVIEW

**Comment by dblock** [comment]
Let's add something to the README about using this and a CHANGELOG entry?

Should there be a workflow that brings up the container in CI?

**Comment by duffn** [comment]
This is nice to show people how to use Docker with grape but if it goes in the repo, feels like it belongs in something like an “examples” directory.

**Comment by ericproulx** [comment]
> This is nice to show people how to use Docker with grape but if it goes in the repo, feels like it belongs in something like an “examples” directory.

@duffn I've added examples in the CONTRIBUTING.md

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -23,6 +23,34 @@ git pull upstream master
...
+
+About the execution process:
+ - displays Ruby, Rubygems, Bundle and Gemfile version when starting:
+    ```
+    ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux-musl]
+    rubygems 3.4.12
+    Bundler version 2.4.1 (2022-12-24 commit f3175f033c)
+    Running default Gemfile
+   ```
+ - keeps the gems to the latest possible version.
```
Comment (dblock) [inline CONTRIBUTING.md:42]: Either remove the period or caputalize the sentence and add it everywhere to keep things consistent.

**Review by dblock — APPROVED** [review]
Nit below.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -23,6 +23,34 @@ git pull upstream master
...
+    ```
+    ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux-musl]
+    rubygems 3.4.12
+    Bundler version 2.4.1 (2022-12-24 commit f3175f033c)
+    Running default Gemfile
+    ```
+ - keeps the gems to the latest possible version
+ - executes under `bundle exec`
+
+Here a some examples:
```
Comment (dblock) [inline CONTRIBUTING.md:45]: Fix English, Here *are* some examples.

Lowercase `running` everywhere to match above.

**Comment by dblock** [comment]
Can I have other maintainers say whether they would like this merged or not?

Should we require a CI to ensure the Dockerfile works?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -23,6 +23,34 @@ git pull upstream master
...
+ - executes under `bundle exec`
+
+Here are some examples:
+
+- running all specs `docker-compose run --rm --build grape rspec`
+- running rspec on a specific file `docker-compose run --rm --build grape rspec spec/:file_path`
+- running task `docker-compose run --rm --build grape rake <task_name>`
+- running rubocop `docker-compose run --rm --build grape rubocop`
+- running all specs on a specific ruby version (e.g 2.7.7) `RUBY_VERSION=2.7.7 docker-compose run --rm --build grape rspec`
+- running specs on a specific gemfile (e.g rails_7.gemfile) `docker-compose run -e GEMFILE=rails_7 --rm --build grape rspec`
```
Comment (dm1try) [inline CONTRIBUTING.md:52]: should be `rails_7_0` now(https://github.com/ruby-grape/grape/commit/cdeaa1329654cca8d074c4b2c091f0ef66a4dbbc)

**Review by dm1try — APPROVED** [review]
I've checked locally, looks good to me)

**Comment by ericproulx** [comment]
rubocop offenses fixed in #2319

**Comment by dblock** [comment]
Thanks. Rebase?

**Pull request merged**